### PR TITLE
Anesthetic kit is a tool that uses anesthesia as charges

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Balance "Anesthetic kit is now a tool, it uses anesthesia as charges, you need one charge for surgery"```

#### Purpose of change

Adresses  #29463
#### Describe the solution

anesthesia kit is a tool made using jar,hose and rags just as before
anesthetic is ammo for the kit, it requires Ether and drugs just as before
anesthetic is liquid

anesthesia "anesthetic kit" moves to obsolete.json
anesthesia is removed from item_groups and doesn't spawn in the game

autodoc uses anesthesia in priority, if the player doesn't have anesthesia the autodoc uses 1 charge of anesthetic from the anesthesia kit.

Install/Uninstall takes 500 moves to perform : the time it takes to setup the autodoc. This is necessary so that the amount of charge remaining in the kit is updated. Save_cyborg is not affected and still uses no move.
Surgery requires the kit with at least 1 charge
#### Describe alternatives you've considered

Makes anesthesia a liquid and remove the kit.
#### Additional context

~~Right now it does not consume ammo as it should.~~

~~I need to figure out how to not break the game too much as it will change current anesthetic kit in anesthesia and players will be required to craft a kit to use anesthesia.
Feedback is appreciated on how to migrate the current system to this one.~~

Migration is handled as follow  : 
 - anesthetic_kit is added to surgery item group with half the probability anesthesia had
 - anesthetic_kit replace anesthesia on every map where it was spawn specifically
 - anesthesia is moved to obsolete.json
 - anesthetic is the ammo for anesthic_kit, it's liquid.
 - anesthetic replace anesthesia in the surgery item group
 - anesthetic spawn in 3L glass jar in 3 portions of 1L
 - anesthetic_kit is auto-learn at firstaid 3. I figured it would be ok since the difficult part is to figure out the quantity of ether and drugs and not to assemble the jar,rag and hoes.

Some side notes : 
- anesthetic is a liquid that looks like ether

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -3962,18 +3962,20 @@ void iexamine::autodoc( player &p, const tripoint &examp )
     }
 
     bool needs_anesthesia = true;
-    std::vector<item_comp> acomps;
+    std::vector< item *> anesth_kit;
```
Comment (ifreund) [inline src/iexamine.cpp]: Why are you not using a `tool_comp` here?

File: src/iexamine.cpp [hunk 2]
```diff
@@ -4040,7 +4042,7 @@ void iexamine::autodoc( player &p, const tripoint &examp )
                 comps.push_back( item_comp( it->typeId(), 1 ) );
                 p.consume_items( comps, 1, is_crafting_component );
                 if( needs_anesthesia ) {
-                    p.consume_items( acomps, 1, is_crafting_component );
+                    anesth_kit[0]->ammo_consume( 1, p.pos() );
```
Comment (ifreund) [inline src/iexamine.cpp]: There's no reason not to use a `tool_comp` and `player::consume_tools()` here.

Comment (Fris0uman) [reply src/iexamine.cpp]: Done

File: src/iexamine.cpp (hunk 1 above)
Comment (Fris0uman) [reply src/iexamine.cpp]: Done

**Comment by Fris0uman** [comment]
Thanks ifreund, now it consumes charges properly.

File: src/iexamine.cpp [hunk 3]
```diff
@@ -3962,18 +3962,20 @@ void iexamine::autodoc( player &p, const tripoint &examp )
     }
 
     bool needs_anesthesia = true;
-    std::vector<item_comp> acomps;
+    std::vector< tool_comp > anesth_kit;
```
Comment (ifreund) [inline src/iexamine.cpp]:
```suggestion
    std::vector<tool_comp> anesth_kit;
```

File: src/iexamine.cpp [hunk 4]
```diff
@@ -3962,18 +3962,20 @@ void iexamine::autodoc( player &p, const tripoint &examp )
...
     if( p.has_trait( trait_NOPAIN ) || p.has_bionic( bionic_id( "bio_painkiller" ) ) ) {
         needs_anesthesia = false;
     } else {
-        std::vector<const item *> a_filter = p.crafting_inventory().items_with( []( const item & it ) {
-            return it.has_flag( "ANESTHESIA" );
+        std::vector< const item *> a_filter = p.crafting_inventory().items_with( []( const item & it ) {
```
Comment (ifreund) [inline src/iexamine.cpp]:
```suggestion
        std::vector<const item *> a_filter = p.crafting_inventory().items_with( []( const item & it ) {
```

File: src/iexamine.cpp [hunk 5]
```diff
@@ -3962,18 +3962,20 @@ void iexamine::autodoc( player &p, const tripoint &examp )
...
+            return it.has_quality( quality_id( "ANESTHESIA" ) );
         } );
         for( const item *anesthesia_item : a_filter ) {
-            acomps.push_back( item_comp( anesthesia_item->typeId(), 1 ) );
+            if( anesthesia_item->ammo_remaining() > 1 ) {
```
Comment (ifreund) [inline src/iexamine.cpp]: Don't *think* you need this check here anymore

Comment (Fris0uman) [reply src/iexamine.cpp]: But if we remove this check where does the "You need _blablabla_ for autodoc to perform any operation." goes ?

Comment (ifreund) [reply src/iexamine.cpp]: Oh my bad, you're totally right.

**Comment by Fris0uman** [comment]
I think it should work like that. Suggestion on what anesthesia should "looks_like" would be good, I left it as first_aid so right now anesthetic_kit and anesthesia look the same (both in tile and in ASCII)

**Comment by kevingranade** [comment]
For migrating, how about:
Keep existing anesthesia item unchanged, but remove it from all spawns and crafts and move to obsolete.json.
Add new anesthesia and anesthesia kit items, anesthesia is a liquid.
Update autodoc to accept both old item and new kit, not just the bare item.

Much later we remove support and definition of old anesthesia item.

**Comment by Fris0uman** [comment]
Ok I'll do that.

File: src/iexamine.cpp [hunk 6]
```diff
@@ -4091,8 +4106,14 @@ void iexamine::autodoc( player &p, const tripoint &examp )
...
                 if( needs_anesthesia ) {
-                    p.consume_items( acomps, 1, is_crafting_component );
+                    if( acomps.empty() ) { // consume obsolete anesthesia first
+                        p.consume_tools( anesth_kit, 1 );
+                    } else {
+                        p.consume_items( acomps, 1, is_crafting_component ); // legacy
+                    }
+
                 }
+                installer.mod_moves(-500);
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]:
```suggestion
                installer.mod_moves( -500 );
```

**Pull request merged**